dds/idl/marshal_generator.cpp

```diff
@@ -2642,6 +2680,29 @@ namespace {
         "Unexpected extensibility while generating MarshalTraits", node);
       return false;
     }
+    be_global->header_ << "; }\n";
+
+    /*
+     * This is used for topic type extensibility level.
+     */
+    const ExtensibilityKind ek = extensibility_kind(struct_node != 0 ? struct_node : union_node);
```

Comment:
```suggestion
    const ExtensibilityKind ek = extensibility_kind(node);
```

Reply:
Have to dynamic cast, but then it would work

Reply:
Okay, I thought `node` was `AST_Type`. It probably should be, but it's fine how it is for now.